router/internal/wsproto/subscriptions_transport_ws.go

```diff
@@ -112,6 +129,9 @@ func (p *subscriptionsTransportWSProtocol) WriteGraphQLErrors(id string, errors
 	if err != nil {
 		return fmt.Errorf("encoding JSON: %w", err)
 	}
+	if err := p.conn.SetWriteDeadline(time.Now().Add(p.writeTimeout)); err != nil {
```

Comment:
Does it make sense to move SetWriteDeadline & SetReadDeadline into conn.WriteJSON & conn.ReadJSON?

Reply:
I think so yes. I'll give it a shot.